code/modules/mob/living/carbon/give.dm

```diff
@@ -38,6 +38,7 @@
 		return
 
 	if(usr.unEquip(I))
-		target.put_in_hands(I) // If this fails it will just end up on the floor, but that's fitting for things like dionaea.
-		usr.visible_message(span("notice", "\The [usr] handed \the [I] to \the [target]."), span("notice", "You give \the [I] to \the [target]."))
-
+		I.on_give()
```

Comment:
You might want to pass the `usr` as `giver` and `target` as `target` to this function. For later extensibility.